pkg/controller/accountclaim/reuse.go

```diff
@@ -185,23 +189,24 @@ func (r *ReconcileAccountClaim) cleanUpAwsAccount(reqLogger logr.Logger, awsClie
 		go cleanUpFunc(reqLogger, awsClient, awsNotifications, awsErrors) //nolint; cleanUpFunc // Not checking return value of goroutine
 	}
 
+	var err error
 	// Wait for clean up functions to end
 	for i := 0; i < len(cleanUpFunctions); i++ {
 		select {
 		case msg := <-awsNotifications:
 			reqLogger.Info(msg)
 		case errMsg := <-awsErrors:
-			err := errors.New(errMsg)
+			err = errors.New(errMsg)
 			reqLogger.Error(err, errMsg)
 			cleanUpStatusFailed = true
 		}
 	}
```

Comment:
I think instead of `var err error` here we need something like `var cleanupErrors[error]{}` and append to that list, and then check if that list is failed.

Otherwise if there are multiple errors we will only see the one error message.

Functionally, this will still get what we're looking for though it might just make future debugging harder in those situations, so I won't block the PR for this request.